sdks/python/apache_beam/io/filebasedsink.py

```diff
@@ -299,6 +323,14 @@ def _rename_batch(batch):
       # May have already been removed.
       pass
 
+  @staticmethod
+  def _template_replace_num_shards(shard_name_template):
+    m = re.search('N+', shard_name_template)
+    if m:
+      shard_name_template = shard_name_template.replace(
+          m.group(0), '%%(num_shards)0%dd' % len(m.group(0)))
```

Comment:
Shouldn't num_shards be the second parameter ?

Reply:
I don't understand this comment.
This code is replacing a substring that looks like `NNN` with `%(num_shards)03d`.

`num_shards` appears in a string in the second parameter to `replace()`.
Where should it be instead?